### PR TITLE
Tree to geo

### PR DESCRIPTION
**Pull request by TheCovenant**

### CODE REVIEW

File: app/geo/gui/src/ts/twxplore/gui/geo/actions/map/ChangeMapFeatureStateAction.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+export const CHANGE_MAP_FEATURE_STATE: CHANGE_MAP_FEATURE_STATE =
+  "CHANGE_MAP_FEATURE_STATE";
+
+export interface ChangeMapFeatureStateAction
+  extends Action<{uris: string[]; state: MapFeatureState}> {
+  type: CHANGE_MAP_FEATURE_STATE;
+}
+
+export const changeMapFeatureState = (
+  a_uri: string[],
```
Comment (gordom6) [inline app/geo/gui/src/ts/twxplore/gui/geo/actions/map/ChangeMapFeatureStateAction.ts]: featureUris

File: app/geo/gui/src/ts/twxplore/gui/geo/actions/map/ChangeMapFeatureStateAction.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  a_state: MapFeatureState
```
Comment (gordom6) [inline app/geo/gui/src/ts/twxplore/gui/geo/actions/map/ChangeMapFeatureStateAction.ts]: newState

File: app/geo/gui/src/ts/twxplore/gui/geo/main.tsx [hunk 3]
```diff
@@ -21,14 +24,17 @@ const browserHistory = createBrowserHistory();
...
-              <Switch>
-                  <Route exact path={Hrefs.home} component={Home}/>
-                  <Route component={NoRoute}/>
-              </Switch>
-          </Router>
-        </LoggerContext.Provider>
+        <Provider store={store}>
+          <LoggerContext.Provider value={logger}>
+            <Router history={browserHistory}>
+                <Switch>
```
Comment (gordom6) [inline app/geo/gui/src/ts/twxplore/gui/geo/main.tsx]: Some people prefer to break these out into a separate component <Routes>. Up to you.

File: app/geo/gui/src/ts/twxplore/gui/geo/reducers/root/rootReducer.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
```
Comment (gordom6) [inline app/geo/gui/src/ts/twxplore/gui/geo/reducers/root/rootReducer.ts]: Don't need this copyright notice.

File: app/geo/gui/src/ts/twxplore/gui/geo/states/map/MapFeature.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { MapFeatureType } from "./MapFeatureType";
+import { MapFeatureState } from "./MapFeatureState";
+
+
+export interface MapFeature {
+  childType?: MapFeatureType;
```
Comment (gordom6) [inline app/geo/gui/src/ts/twxplore/gui/geo/states/map/MapFeature.ts]: Will probably want to wrap the API Feature or extend it. Do that next PR.

File: app/geo/gui/src/ts/twxplore/gui/geo/states/map/MapFeatureType.ts [hunk 6]
```diff
@@ -0,0 +1,6 @@
+export enum MapFeatureType {
```
Comment (gordom6) [inline app/geo/gui/src/ts/twxplore/gui/geo/states/map/MapFeatureType.ts]: This file should go away,  you'll get a type from the API.

File: app/geo/gui/src/ts/twxplore/gui/geo/store.ts [hunk 7]
```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
```
Comment (gordom6) [inline app/geo/gui/src/ts/twxplore/gui/geo/store.ts]: No copyright notice.

File: app/geo/gui/types/graphql.d.ts [hunk 8]
```diff
@@ -4,3 +4,13 @@ declare module "*.graphql" {
 
   export = Schema;
 }
+
+declare module 'kepler.gl'
```
Comment (gordom6) [inline app/geo/gui/types/graphql.d.ts]: Move these somewhere else, types/kepler.d.ts, types/global.d.ts, types/react-palm.d.ts.

File: app/tree/gui/package.json [hunk 9]
```diff
@@ -7,7 +7,9 @@
     "@material-ui/icons": "^4.9.1",
     "@material-ui/lab": "^4.0.0-alpha.44",
     "@tetherless-world/twxplore-base-lib": "1.0.1",
+    "@types/graphql": "^14.5.0",
```
Comment (gordom6) [inline app/tree/gui/package.json]: Not sure why this was modified?

Comment (TheCovenant) [reply app/tree/gui/package.json]: Probably something I did in desperation while fixing type/import errors before copying in graphql.d.ts

**Pull request merged**